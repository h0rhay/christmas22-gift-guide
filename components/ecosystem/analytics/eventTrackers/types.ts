import { ClickEventActions } from "./eventsConstants";
import { EventTrackerType } from "./eventTracker";

export type GoogleAnalyticsType = {
  ga_event_name: string;
  ga_eventcategory?: string;
  ga_eventaction?: ClickEventActions;
  ga_eventlabel?: string;
};

export type CommonEventsTypes = {
  page_name: string;
  page_type?: string;
}

export type ViewEventTrackerType = {
  // specific for view event
  pageload_event?: string;
  page_load_time?: string; // not used anymore?
  page_subtype?: string;
  page_breadcrumb?: string
  page_category_name?: string
} & CommonEventsTypes
  & GoogleAnalyticsType
  & EventTrackerType

export type LinkEventTrackerType =
  CommonEventsTypes
  & GoogleAnalyticsType
  & EventTrackerType
  & Omit<EventTrackerType, 'event_name' | 'ga_event_name' | 'ga_eventcategory' | 'ga_eventaction'>;

export type DynamicViewEventTrackerType = {
  page_name: string;
}

export type DynamicLinkEventTrackerType = {
  ga_eventaction?: ClickEventActions;
  page_name: string;
  ga_eventlabel?: string;
}
