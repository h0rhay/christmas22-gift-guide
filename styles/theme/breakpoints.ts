export const REM = 16;

export const dimensionsInPx = {
  small: 960,
  large: 961,
};

export const breakpoints = {
  small: `@media (max-width: ${dimensionsInPx.small / REM}rem)`,
  large: `@media (min-width: ${dimensionsInPx.large / REM}rem)`,
};

const SLIDER_TABLET = 1080;
const SLIDER_MOBILE = 576;

export const tablet = `@media only screen 
and (min-width: ${SLIDER_MOBILE}px) 
and (max-width: ${SLIDER_TABLET}px) `;

export const tabletAndMobile = `@media (max-width: ${SLIDER_TABLET}px) `;
export const mobile = `@media (max-width: ${SLIDER_MOBILE}px) `;
