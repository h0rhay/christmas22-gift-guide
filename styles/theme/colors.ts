import { PrimitiveColorsType } from "types";

const colors: PrimitiveColorsType = {
  red: "rgba(229,58,52,1);",
  bubblegum: "rgba(242,188,200,1);",
  olive: "rgba(87,93,43,1);",
  yolk: "rgba(255,194,92,1);",
  sfYellow: "rgba(255,226,85,1);",
  milk: "rgba(254,251,240,1);",
  milkTransparent: "rgba(254,251,240,0.8);",
  whiteTransparent: "rgba(255, 255, 255, 0.8);",
  black: "rgba(33, 33, 33, 1)",
  white: "rgba(255, 255, 255, 1)",
  grey: {
    main: "rgba(184,184,184,1);",
    mid: "rgba(234,234,234,1);",
    light: "rgba(246,246,246,1);",
  },
};

export default colors;
