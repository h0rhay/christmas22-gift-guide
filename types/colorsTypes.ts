export type ColorGroupType = {
  surface: string;
  interface: string;
};

export type PrimitiveGreyColorType = {
  main: string;
  mid: string;
  light: string;
};

export type PrimitiveColorsType = {
  red: string;
  bubblegum: string;
  olive: string;
  yolk: string;
  sfYellow: string;
  milk: string;
  grey: PrimitiveGreyColorType;
};
