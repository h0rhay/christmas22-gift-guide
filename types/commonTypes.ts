export type CommonSizesType = {
  tinyGap?: "0.5rem";
  extraSmallGap?: string;
  smallGap?: string;
  mediumGap?: string;
  largeGap?: string;
  small: string;
  medium: string;
  large: string;
  extraSmall?: string;
  xlarge?: string;
};

export type LineHeight = CommonSizesType & {
  heading: string;
};

const commonSizesEntryItems = [
  "small",
  "medium",
  "large",
  "xlarge",
  "extraSmall",
] as const;
export type CommonSizesEntriesType = typeof commonSizesEntryItems[number];
