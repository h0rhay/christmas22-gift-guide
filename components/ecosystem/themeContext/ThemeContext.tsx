import React, { ReactChild, ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import theme from "styles/theme";

const ColoredSurface = styled.div`
  background-color: ${({ theme }) => theme.currentColorGroup.surface};
  * {
    color: ${({ theme }) => theme.currentColorGroup.interface};
  }
`;

type ThemeContextProps = {
  colorGroup?: string;
  children: ReactChild | Array<ReactChild> | ReactNode;
};

const ThemeContext = ({
  children,
  colorGroup = theme.fallbackColorGroup,
}: ThemeContextProps) => {
  const currentColorGroup =
    theme.colorsGroup[colorGroup] ||
    theme.colorsGroup[theme.fallbackColorGroup];
  return (
    <ThemeProvider theme={{ ...theme, currentColorGroup }}>
      <ColoredSurface className="colored-surface">{children}</ColoredSurface>
    </ThemeProvider>
  );
};

export default ThemeContext;
