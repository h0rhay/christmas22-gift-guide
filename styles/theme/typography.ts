import {
  FontFamilyType,
  FontSizeType,
  FontStyleType,
  FontWeightType,
  TypographyType,
} from "types";
import { CommonSizesType, LineHeight } from "../../types/commonTypes";

const fontFamily: FontFamilyType = {
  formula: '"Formula-Condensed",sans-serif',
  apoc: "Apoc-Revelations",
  universalSansDisplay: "Universal-Sans",
  avalonDemi: "Avalon-Demi",
  default: "-apple-system, BlinkMacSystemFont, Helvetica, Arial, sans-serif",
};

const fontWeight: FontWeightType = {
  light: 300,
  regular: 400,
  medium: 500,
  bold: 700,
};

const fontWeightEntryItems = ["light", "regular", "medium", "bold"] as const;
export type fontWeightEntriesType = typeof fontWeightEntryItems[number];

const fontSize: FontSizeType = {
  small: "1rem",
  medium: "1.25rem",
  large: "1.5rem",
  xlarge: "2rem",
};

const headingFontSize: CommonSizesType = {
  // small: "clamp(1.424rem, 1.75vw, 1.75rem)",
  small: "clamp(1.3rem, calc(1.3rem + ((1vw - 7.68px) * 0.6944)), 1.75rem)",
  medium: "clamp(2rem, calc(2rem + ((1vw - 7.68px) * 0.3472)), 2.25rem);",
  large: "clamp(2.3rem, calc(2.3rem + ((1vw - 7.68px) * 1.3889)), 4rem)",
  xlarge: "clamp(4rem, calc(4rem + ((1vw - 7.68px) * 2.7778)), 6rem)",
};

const lineHeight: LineHeight = {
  heading: "1.1",
  small: "1.2",
  medium: "1.5",
  large: "1.75",
};

const letterSpacing: CommonSizesType = {
  small: "1px",
  medium: "2px",
  large: "3px",
};

const fontStyle: FontStyleType = {
  normal: "normal",
  italic: "italic",
};

const fontStyleEntryItems = ["normal", "italic"] as const;
export type fontStyleEntriesType = typeof fontStyleEntryItems[number];

const space: CommonSizesType = {
  tinyGap: "0.5rem",
  extraSmallGap: "2rem",
  smallGap: "2.5rem",
  mediumGap: "4rem",
  largeGap: "5rem",
  extraSmall: ".5rem",
  small: "1rem",
  medium: "1.5rem",
  large: "2rem",
};

const typography: TypographyType = {
  fontFamily,
  fontWeight,
  fontSize,
  headingFontSize,
  lineHeight,
  letterSpacing,
  fontStyle,
  space,
};

export default typography;
