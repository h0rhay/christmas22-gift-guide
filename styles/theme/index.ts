import { breakpoints } from "./breakpoints";
import typography from "./typography";
import milkColorGroup from "./colorGroups/milk";
import bubblegumColorGroup from "./colorGroups/bubblegum";
import yolkColorGroup from "./colorGroups/yolk";
import oliveColorGroup from "./colorGroups/olive";
import heroColorGroup from "./colorGroups/hero";
import darkColorGroup from "./colorGroups/dark";
import lightColorGroup from "./colorGroups/light";
import colors from "styles/theme/colors";
import { TypographyType, PrimitiveColorsType, ColorsGroupType } from "types";
import { ColorGroupType } from "types/colorsTypes";

const colorsGroup = {
  hero: heroColorGroup,
  milk: milkColorGroup,
  olive: oliveColorGroup,
  bubblegum: bubblegumColorGroup,
  yolk: yolkColorGroup,
  dark: darkColorGroup,
  light: lightColorGroup,
};

const colorsInputs = [
  "hero",
  "milk",
  "bubblegum",
  "yolk",
  "olive",
  "dark",
  "light",
] as const;
export type ColorGroupInputType = typeof colorsInputs[number];

export type ThemeType = {
  typography: TypographyType;
  breakpoints: any; // TODO
  colors: PrimitiveColorsType;
  colorsGroup: ColorsGroupType;
  colorGroupNames: ColorGroupInputType[];
  fallbackColorGroup: ColorGroupInputType;
  currentColorGroup: ColorGroupType;
};

const theme: ThemeType = {
  typography,
  breakpoints,
  colors,
  colorsGroup,
  colorGroupNames: Object.keys(colorsGroup) as ColorGroupInputType[],
  fallbackColorGroup: "milk", //Consider Object.keys(colorsGroup)[0];
  currentColorGroup: heroColorGroup, // dynamic changes
};

export default theme;
