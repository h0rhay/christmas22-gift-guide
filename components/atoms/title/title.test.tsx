import { render } from "@testing-library/react";
import Title from "components/atoms/title/title";
import { ThemeProvider } from "styled-components";
import theme from "styles/theme";
import ThemeContext from "../../ecosystem/themeContext/ThemeContext";
import { CommonSizesEntriesType } from "types/commonTypes";

type renderTitleComponentType = {
  colorGroupName: string;
  titleProps?: {
    isBold?: boolean;
    isCenter?: boolean;
    size?: CommonSizesEntriesType;
  };
};

const defaultRenderTitleComponentProps = {
  colorGroupName: theme.colorGroupNames[0] as string,
  titleProps: {},
};

const TITLE_SELECTOR_CLASS_NAME = "xmas-title";

const renderComponent = ({
  colorGroupName,
  titleProps,
}: renderTitleComponentType = defaultRenderTitleComponentProps) =>
  render(
    <ThemeProvider theme={theme}>
      <ThemeContext colorGroup={colorGroupName}>
        <Title
          isBold={titleProps?.isBold}
          size={titleProps?.size}
          isCenter={titleProps?.isCenter}
        >
          Sample test text for title
        </Title>
      </ThemeContext>
    </ThemeProvider>
  );

describe("Title", () => {
  it("Square should match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("Title should match default params for fontSize=headingFontSize.large, fontWeight=fontWeight.light and color=colorGroup.interface and text-align=center", () => {
    const colorGroupName = theme.colorGroupNames[0];
    const colorGroup = theme.colorsGroup[colorGroupName as string];
    const { container } = renderComponent();

    const titleEl = container.getElementsByClassName(
      TITLE_SELECTOR_CLASS_NAME
    )[0];

    expect(titleEl).toHaveStyle(`color: ${colorGroup?.interface}`);
    expect(titleEl).toHaveStyle("font-size: 1.5em;");
    expect(titleEl).toHaveStyle(
      `font-weight: ${theme.typography.fontWeight.light}`
    );
    expect(titleEl).toHaveStyle("text-align: center");
    expect(titleEl).toHaveStyle("text-transform: uppercase");
    expect(titleEl).toHaveStyle("font-style: normal");
  });

  it("Title should always match with currentColorGroup.interface color", () => {
    let colorGroupName = theme.colorGroupNames[0] as string;
    let colorGroup = theme.colorsGroup[colorGroupName as string];
    let { container } = renderComponent({ colorGroupName });

    expect(
      container.getElementsByClassName(TITLE_SELECTOR_CLASS_NAME)[0]
    ).toHaveStyle(`color: ${colorGroup?.interface}`);
    colorGroupName = theme.colorGroupNames[1] as string;
    colorGroup = theme.colorsGroup[colorGroupName as string];

    ({ container } = renderComponent({ colorGroupName }));
    expect(
      container.getElementsByClassName(TITLE_SELECTOR_CLASS_NAME)[0]
    ).toHaveStyle(`color: ${colorGroup?.interface}`);
  });

  it("Title should ovewrite fontWeight with isBold param", () => {
    const colorGroupName = theme.colorGroupNames[0] as string;
    const { container } = renderComponent({
      colorGroupName,
      titleProps: { isBold: true },
    });

    expect(
      container.getElementsByClassName(TITLE_SELECTOR_CLASS_NAME)[0]
    ).toHaveStyle(`font-weight: ${theme.typography.fontWeight.bold}`);
  });

  it("Title should ovewrite textAlign center with isCenter param", () => {
    const colorGroupName = theme.colorGroupNames[0] as string;
    const { container } = renderComponent({
      colorGroupName,
      titleProps: { isCenter: false },
    });

    expect(
      container.getElementsByClassName(TITLE_SELECTOR_CLASS_NAME)[0]
    ).not.toHaveStyle("text-align: center");
  });
});
