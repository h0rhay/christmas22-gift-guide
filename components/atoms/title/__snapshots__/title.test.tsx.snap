// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Square should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <h2
      class="title__StyledTitle-sc-z5nezq-0 gOeEph xmas-title "
    >
      Sample test text for title
    </h2>
  </div>
</div>
`;
